### DISCUSSION TITLE
xc-token for API key [Jun 16]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2475] [Discord Threads]
> 🙋‍♂️ **arsenio**

**Comment by o1lab** [comment]
> 💁 **o1lab**  
#### xc-auth is meant only for GUI usage and has a refresh time (10 or 12 hrs)  
#  
> 💁 **o1lab**  
#### use xc-token header  
#  
> 💁 **o1lab**  
#### create an API token within the prodcut  
#  
> 🙋‍♂️ **wkw**  
#### pls take a look here.    
####    
#### https://docs.nocodb.com/developer-resources/accessing-apis  
#  
> 🙋‍♂️ **arsenio**  
#### yes, I did it. there is no such info. My question is If I can restore the old API  
#  
> 🙋‍♂️ **arsenio**  
#### before the update I was using the same API for months  
#  
> 🙋‍♂️ **arsenio**  
#### I mean xc-auth. but now probably I need to change in all my apps to xc-token, right?  
#  
> 💁 **o1lab**  
#### xc-token is an API token that you can use until you delete the token  
#  
> 💁 **o1lab**  
#### xc-auth has a expiry date  
#  
> 💁 **o1lab**  
#### unsure on how u managed to use xc-auth without refresh  
#  
> 🙋‍♂️ **arsenio**  
#### all right. thanx.  
#  
> 🙋‍♂️ **VA-Ohm**  
#### same here, I made one (fortunately only one) script and it used xc-auth for several months  
#  
> 🙋‍♂️ **VA-Ohm**  
#### just today I got the error  
#  
> 🙋‍♂️ **VA-Ohm**  
#### it has never expired before  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/986544848922345512)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
